File: src/AniTracker.Api/Models/UserMedia.cs
namespace AniTracker.Api.Models;

public class UserMedia
{
    public required Guid Id { get; set; }
    public required Guid UserId { get; set; }
    public required Guid MediaId { get; set; }
    public required WatchStatus Status { get; set; }
    public required int? Rating { get; set; }
    public required int EpisodesWatched { get; set; }
    public required DateTime? StartDate { get; set; }
    public required DateTime? CompletedDate { get; set; }
    public required bool IsFavorite { get; set; }
}

File: src/AniTracker.Frontend/src/models/userMedia.ts
import type WatchStatus from "@/enums/watchStatus";

type UserMedia = {
    watchStatus: WatchStatus;
    id: string;
    userId: string;
    mediaId: string;
    rating: number | null;
    episodesWatched: number;
    startDate: Date | null;
    completedDate: Date | null;
    isFavorite: boolean;
}

export default UserMedia;